packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -1668,20 +1666,53 @@ void main() {
       }
 
       await tester.pumpWidget(buildFrame(TextDirection.ltr));
-      await expectLater(
-        find.byType(InputDecorator),
-        matchesGoldenFile('m3_input_decorator.outline_label.ltr.png'),
+      RenderBox borderBox = InputDecorator.containerOf(tester.element(findBorderPainter()))!;
+      // Convert label bottom left offset to border path coordinate system.
+      final Offset labelBottomLeftLocalToBorder = borderBox.globalToLocal(getLabelRect(tester).bottomLeft);
+
+      expect(findBorderPainter(), paints
+        ..save()
+        ..path(
+          // The label bottom left corner should not be part of the border.
+          excludes: <Offset>[
+            labelBottomLeftLocalToBorder,
+          ],
+          // The points just before the label bottom left corner should be part of the border.
+          includes: <Offset>[
+            labelBottomLeftLocalToBorder - const Offset(1, 0),
+            labelBottomLeftLocalToBorder - const Offset(1, 1),
+          ],
+        )
+        ..restore(),
       );
 
       await tester.pumpWidget(buildFrame(TextDirection.rtl));
-      await expectLater(
-        find.byType(InputDecorator),
-        matchesGoldenFile('m3_input_decorator.outline_label.rtl.png'),
+      borderBox = InputDecorator.containerOf(tester.element(findBorderPainter()))!;
+      // Convert label bottom right offset to border path coordinate system.
+      Offset labelBottomRightLocalToBorder = borderBox.globalToLocal(getLabelRect(tester).bottomRight);
+      // TODO(bleroux): determine why the position has to be moved by 2 pixels to the right.
+      // See https://github.com/flutter/flutter/issues/150109.
```

Comment:
It's especially weird because the screenshot in your issue makes it look like the border is clearly several pixels away from the bottom right of the label. Could be something like a rounding thing as you suggested. I think I remember the code that does the break in the border is not super precise.